Sources/Basic/TerminalController.swift

```diff
@@ -0,0 +1,252 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import libc
+import func POSIX.getenv
+
+private protocol OutputFileStreamProtocol: OutputStream {
+    var fd: Int32 { get }
+    var file: UnsafeMutablePointer<FILE> { get }
+}
+
+final public class OutputFileStream: OutputFileStreamProtocol {
+    private let fd: Int32
+    private let file: UnsafeMutablePointer<FILE>
+
+    init(file: UnsafeMutablePointer<FILE>) {
+        self.file = file
+        self.fd = fileno(file)
+    }
+
+    public func write(_ string: String) {
+        _ = fputs(string, file)
+    }
+}
+
+public var stdout = OutputFileStream(file: libc.stdout)
+public var stderr = OutputFileStream(file: libc.stderr)
+
+/// Terminal color choices.
+public enum TermColor {
+    case noColor
+
+    case red
+    case green
+    case yellow
+
+    case white
+    case black
+    case grey
+
+    /// Returns the color code which can be prefixed on a string to display it in that color.
+    /// Note: To avoid unexpected behavior the color codes are private and not exposed to clients
+    /// other than TerminalController.
+    private var string: String {
+        switch self {
+        case .noColor: return ""
+        case .red: return "\u{001B}[31m"
+        case .green: return "\u{001B}[32m"
+        case .yellow: return "\u{001B}[33m"
+        case .white: return "\u{001B}[37m"
+        case .black: return "\u{001B}[30m"
+        case .grey: return "\u{001B}[30;1m"
+        }
+    }
+}
+
+/// Number of the columns in the current terminal.
+// FIXME: What if user resizes terminal during execution?
```

Comment:
I _believe_ there is a way to get notified of that via a signal, but I've never written code to handle it.